### PR TITLE
Repo set-up (primarily documentation and CI/CD files)

### PR DESCRIPTION
**Pull request by mary-cleaton**
## Description
This pull request introduces various set-up files. These are primarily documentation (e.g. `README.md`, `CODEOWNERS`, `CONTRIBUTORS.md`, `branch_and_deploy_guide.md`) and CI/CD files (e.g. pre-commits, GitHub Actions). Many are copied from `ONSdigital/rdsa-utils` with minor or no edits.

Overall, the pull request aims to:
- Ensure compliance with [ONSdigital GitHub policy](https://github.com/ONSdigital/ons-template/wiki/).
- Provide guidance for maintainers, developers and contributors.
- Enforce good practice via pre-commits and GitHub Actions.

It does not aim to:
- Introduce CI/CD for package creation.
- Introduce CI/CD for creation of a GitHub webpage.

These will be introduced in future documentation and CI/CD file updates.

## Type of change

- [ ] Bug fix - *non-breaking change*
- [ ] New feature - *non-breaking change*
- [ ] Breaking change - *backwards incompatible change, changes expected behaviour*
- [x] Non-user facing change, structural change, dev functionality, docs ...

## Checklist:

- [x] I have performed a self-review of my own code.
- [ ] I have commented my code appropriately, focusing on explaining my design decisions (explain why, not how).
- [ ] I have made corresponding changes to the documentation (comments, docstring, etc.. )
- [ ] I have added tests that prove my fix is effective or that my feature works.
- [ ] New and existing unit tests pass locally with my changes.
- [x] I have updated the change log.

<br>

#  Peer review
Any new code includes all the following:

- **Documentation**: docstrings, comments have been added/ updated.
- **Style guidelines**: New code conforms to the project's contribution guidelines.
- **Functionality**: The code works as expected, handles expected edge cases and exceptions are handled appropriately.
- **Complexity**: The code is not overly complex, logic has been split into appropriately sized functions, etc..
- **Test coverage**: Unit tests cover essential functions for a reasonable range of inputs and conditions. Added and existing tests pass on my machine.

### Review comments
Suggestions should be tailored to the code that you are reviewing. Provide context.
Be critical and clear, but not mean. Ask questions and set actions.
<details><summary>These might include:</summary>

- bugs that need fixing (does it work as expected? and does it work with other code
  that it is likely to interact with?)
- alternative methods (could it be written more efficiently or with more clarity?)
- documentation improvements (does the documentation reflect how the code actually works?)
- additional tests that should be implemented
  - Do the tests effectively assure that it
  works correctly? Are there additional edge cases/ negative tests to be considered?
- code style improvements (could the code be written more clearly?)
</details>
<br>

*Further reading: [code review best practices](https://best-practice-and-impact.github.io/qa-of-code-guidance/peer_review.html)*

### CODE REVIEW

File: .github/workflows/pull_request_workflow.yaml [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+  
+  # Enforces the running of the pre-commit hooks on every pull request
+  pre-commit:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Check out repository
+        uses: actions/checkout@v3
+      
+      - name: Set up Python
+        uses: actions/setup-puython@v4
```
[the remark this answers is not in the document]
Comment (mary-cleaton) [reply .github/workflows/pull_request_workflow.yaml]: Fixed this, but CI checks still failing.

**Review by grthomson — DISMISSED** [review]
All checks passing, only issues remaining are minor typos in branch_and_deploy_guide.md

I think that if I approve now, it will be revoked when typos are fixed (branch rule "Dismiss stale pull request approvals when new commits are pushed"). Can test this. :)

**Comment by mary-cleaton** [comment]
Have fixed all typos noted above.
